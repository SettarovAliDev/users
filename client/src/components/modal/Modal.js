import { Fragment } from "react";
import ReactDOM from "react-dom";

import { StyledBackdrop, StyledModal } from "./ModalStyles";

const portalElement = document.getElementById("overlays");

const Modal = (props) => {
  return (
    <Fragment>
      {ReactDOM.createPortal(<StyledBackdrop />, portalElement)}
      {ReactDOM.createPortal(
        <StyledModal>{props.children}</StyledModal>,
        portalElement
      )}
    </Fragment>
  );
};

export default Modal;
