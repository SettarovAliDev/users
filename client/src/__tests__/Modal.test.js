import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { cleanup } from "@testing-library/react";
import "jest-styled-components";

import Modal from "../components/modal/Modal";

beforeAll(() => {
  ReactDOM.createPortal = jest.fn((element, node) => {
    return element;
  });
});

afterEach(cleanup);
afterEach(() => {
  ReactDOM.createPortal.mockClear();
});

it("renders modal", () => {
  const component = renderer.create(
    <Modal>
      <button>BUTTON</button>
    </Modal>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
