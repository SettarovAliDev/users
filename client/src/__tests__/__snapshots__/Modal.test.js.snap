// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders modal 1`] = `null`;
